Forge.TreeWalker.UnitTests/test/TreeSchemaValidatorTests.cs

```diff
@@ -0,0 +1,166 @@
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Schema;
+    using Microsoft.Forge.TreeWalker;
+
+    [TestClass]
+    public class TreeSchemaValidatorTests
+    {
+        private string forgeTreeAsString;
+        private ForgeTree treeSchema;
+        private Dictionary<string, ForgeTree> treeSchemas;
+        private string forgeTreeFromPath;
+        private string schemaDirectoryPath;
+        private string stringRules;
+        private string rulesForDictionary;
+        private JSchema jschemaRules;
+        private string invalidSchemaDirectoryPath;
+        private string invalidSchemaWithErrorContent;
+        private string forgeTreeDictionaryAsString;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            forgeTreeFromPath = Path.Combine(Environment.CurrentDirectory, "test\\ExampleSchemas\\TardigradeSchema.json");
+            schemaDirectoryPath = Path.Combine(Environment.CurrentDirectory, "test\\ExampleSchemas");
+            stringRules = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "contracts\\ForgeSchemaValidationRules.json"));
+            rulesForDictionary = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "contracts\\ForgeSchemaDictionaryValidationRules.json"));
+            jschemaRules = JSchema.Parse(stringRules);
+            forgeTreeAsString = File.ReadAllText(forgeTreeFromPath);
+            forgeTreeDictionaryAsString = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "test\\ExampleSchemas\\SubroutineSchema.json"));
+            treeSchema = JsonConvert.DeserializeObject<ForgeTree>((string)forgeTreeAsString);
+            treeSchemas = new Dictionary<string, ForgeTree>();
+            treeSchemas.Add("tree1", treeSchema);
+            treeSchemas.Add("tree2", treeSchema);
+            treeSchemas.Add("tree3", treeSchema);
+            invalidSchemaWithErrorContent = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "test\\InvalidTestSchemas\\InvalidTestSchemaErrorContent.json"));
+            invalidSchemaDirectoryPath = "test\\ExampleSchemas\\TardigradeSchema.json";
+        }
+
+        [TestMethod]
+        public void Test_GetLinkedJSchemaRules()
+        {
+            try
+            {
+                JSchema linkedRules = ForgeSchemaValidator.GetLinkedJSchemaRules(stringRules, stringRules, "//ForgeSchemaValidationRules.json");
+                ForgeSchemaValidator.ValidateSchemaAsForgeTree(treeSchema, linkedRules, out IList<ValidationError> errorList);
+            }
+            catch (Exception ex)
+            {
+                Assert.Fail("Expected no exception, but got: " + ex.Message);
+            }
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaAsForgeTree_WithRulesAsString()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsForgeTree(treeSchema, stringRules, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
```

Comment:
You can change all these to Assert.IsTrue instead of Assert.AreEqual(true, res).

Assert.IsTrue(ForgeSchemaValidator.ValidateSchemaAsForgeTree(treeSchema, stringRules, out IList<ValidationError> errorList));
Assert.AreEqual(0, errorList.Count);